cumulusci/tasks/connectedapp.py

```diff
@@ -0,0 +1,174 @@
+    def _init_options(self, kwargs):
+        self.client_id = None
+        self.client_secret = None
+        kwargs["command"] = "force:mdapi:deploy --wait 5"
+        super(CreateConnectedApp, self)._init_options(kwargs)
+
+        # Validate label
+        if not re.match(r"^\w+$", self.options["label"]):
+            raise TaskOptionsError(
+                "label value must contain only alphanumeric or underscore characters"
+            )
+
+        # Default email to the github service's email if configured
+        if "email" not in self.options:
+            try:
+                github = self.project_config.keychain.get_service("github")
+            except ServiceNotConfigured:
+                raise TaskOptionsError(
+                    "Could not determine a default for option 'email'.  Either configure the github service using 'cci service connect github' or provide a value for the 'email' option"
+                )
+            self.options["email"] = github.email
+
+        # Default to sfdx defaultdevhubusername
+        if "username" not in self.options:
+            self._set_default_username()
+        self.options["command"] += " -u {}".format(self.options.get("username"))
+
+        self.options["connect"] = process_bool_arg(self.options.get("connect", False))
+        self.options["overwrite"] = process_bool_arg(
+            self.options.get("overwrite", False)
+        )
+
+    def _set_default_username(self):
+        self.logger.info("Getting username for the default devhub from sfdx")
+        self._run_command(
+            command="{} force:config:get defaultdevhubusername --json".format(SFDX_CLI),
+            env=self._get_env(),
+            output_handler=self._process_devhub_output,
+        )
+
+    def _process_json_output(self, line):
+        try:
+            data = json.loads(line)
+            return data
+        except Exception:
+            self.logger.error("Failed to parse json from line: {}".format(line))
+            raise
+
+    def _process_devhub_output(self, line):
+        data = self._process_json_output(line)
+        if "value" not in data["result"][0]:
+            raise TaskOptionsError(
+                "No sfdx config found for defaultdevhubusername.  Please use the sfdx force:config:set to set the defaultdevhubusername and run again"
+            )
+        self.options["username"] = data["result"][0]["value"]
+
+    def _generate_id_and_secret(self):
+        self.client_id = random_alphanumeric_underscore(85)
+        self.client_secret = random_alphanumeric_underscore(32)
+
+    def _build_package(self):
+        connected_app_path = os.path.join(self.tempdir, "connectedApps")
+        os.mkdir(connected_app_path)
+        self._generate_id_and_secret()
+        with open(
+            os.path.join(connected_app_path, self.options["label"] + ".connectedApp"),
+            "w",
+        ) as f:
+            f.write(
+                CONNECTED_APP.format(
+                    **{
```

Comment:
instead of a splatted dict, you can just pass in kwargs, e.g.

```
CONNECTED_APP.format(label = self.option["label"], email = self.options["email"])
```
etc...